crates/timers/src/future.rs

```diff
@@ -71,37 +54,32 @@ impl TimeoutFuture {
     /// # Example
     ///
     /// ```no_run
-    /// # extern crate futures_rs as futures;
-    /// use futures::prelude::*;
     /// use gloo_timers::future::TimeoutFuture;
+    /// use wasm_bindgen_futures::spawn_local;
     ///
-    /// wasm_bindgen_futures::spawn_local(
-    ///     TimeoutFuture::new(1_000).map(|_| {
-    ///         // Do stuff after one second...
-    ///     })
-    /// );
+    /// spawn_local(async {
+    ///     TimeoutFuture::new(1_000).await;
+    ///     // Do stuff after one second...
+    /// });
     /// ```
     pub fn new(millis: u32) -> TimeoutFuture {
-        let mut id = None;
-        let promise = js_sys::Promise::new(&mut |resolve, _reject| {
-            id = Some(set_timeout(&resolve, millis as i32));
+        let (tx, rx) = oneshot::channel();
+        let inner = Timeout::new(millis, move || {
+            // if the receiver was dropped we do nothing.
+            let _ = tx.send(());
```

Comment:
I think this should `unwrap_throw`, just as an extra precaution.

Reply:
Yep you're right if the receiver is dropped then the whole thing is dropped so this closure should never run.